src/community/label_propagation.c

```diff
@@ -213,163 +639,35 @@ igraph_error_t igraph_community_label_propagation(const igraph_t *graph,
         }
     } else {
         for (i = 0; i < no_of_nodes; i++) {
-            VECTOR(*membership)[i] = i + 1;
+            VECTOR(*membership)[i] = i;
         }
     }
 
-    reversed_mode = IGRAPH_REVERSE_MODE(mode);
-
     /* From this point onwards we use 'fixed_copy' instead of 'fixed' */
+    IGRAPH_VECTOR_INIT_FINALLY(&label_counters, no_of_nodes);
 
-    /* Create an adjacency/incidence list representation for efficiency.
-     * For the unweighted case, the adjacency list is enough. For the
-     * weighted case, we need the incidence list */
-    if (weights) {
-        IGRAPH_CHECK(igraph_inclist_init(graph, &il, reversed_mode, IGRAPH_LOOPS_ONCE));
-        IGRAPH_FINALLY(igraph_inclist_destroy, &il);
-    } else {
-        IGRAPH_CHECK(igraph_adjlist_init(graph, &al, reversed_mode, IGRAPH_LOOPS_ONCE, IGRAPH_MULTIPLE));
-        IGRAPH_FINALLY(igraph_adjlist_destroy, &al);
-    }
+    switch(lpa_variant)
+    {
+    case IGRAPH_LPA_FAST:
+        IGRAPH_CHECK(igraph_i_community_fast_label_propagation(graph, membership, mode, weights, fixed_copy));
+        break;
 
-    /* Create storage space for counting distinct labels and dominant ones */
-    IGRAPH_VECTOR_INIT_FINALLY(&label_counters, no_of_nodes + 1);
-    IGRAPH_VECTOR_INT_INIT_FINALLY(&dominant_labels, 0);
-    IGRAPH_VECTOR_INT_INIT_FINALLY(&nonzero_labels, 0);
-    IGRAPH_CHECK(igraph_vector_int_reserve(&dominant_labels, 2));
+    case IGRAPH_LPA_RETENTION:
+        IGRAPH_CHECK(igraph_i_community_label_propagation(graph, membership, mode, weights, fixed_copy, /* retention */ true ));
+        break;
 
-    /* Initialize node ordering vector with only the not fixed nodes */
-    if (fixed_copy) {
-        IGRAPH_VECTOR_INT_INIT_FINALLY(&node_order, no_of_not_fixed_nodes);
-        for (i = 0, j = 0; i < no_of_nodes; i++) {
-            if (!VECTOR(*fixed_copy)[i]) {
-                VECTOR(node_order)[j] = i;
-                j++;
-            }
-        }
-    } else {
-        IGRAPH_CHECK(igraph_vector_int_init_range(&node_order, 0, no_of_nodes));
-        IGRAPH_FINALLY(igraph_vector_int_destroy, &node_order);
+    case IGRAPH_LPA_DOMINANCE:
+    default:
```

Comment:
The `default` label should be separate and throw an error (incorrect LPA variant).

Reply:
Sure, we can do that.